terminal/linux_macos-terminal/README.md

```diff
@@ -7,19 +7,68 @@ The challenge with teaching anyone how to use the terminal is the rather large l
-Example 1: `rm thisisatest.txt` -> this removes the text file you created with `touch` in the last example
-Example 2: `rm -r this_is_a_folder` -> the `-r` flag means "recursive", and is necessary to delete folders. This command deletes the folder you created in a previous example. 
+- **pwd**: "Print Working Directory". This command prints your current position within your filesystem.
+    -  Example: `$pwd` -> `/Users/kushdesai/github/ultimate-hackathon-starting-guide/getting-started-terminal/linux_macos-terminal`
+- **cd**: Stands for "change directory". Used to navigate into folders
+    - Example: `cd ../windows-terminal` -> no output, but I am not in the windows-terminal folder. `..` refers to moving back ("up") one directory
+- **ls**: "list". This command lists all the files present in the current folder.
+    - Example: `ls` -> `README.md`, since this README is the only folder in the linux_macos-terminal folder. 
+- **touch**: Creates a file
+    - Example: `touch thisisatest.txt` -> creates a file named thisisatest.txt in the current directory
+- **mkdir**: Makes a directory (folder)
+    - Example: `mkdir this_is_a_folder` -> creates a folder named this_is_a_folder in the current directory. 
+- **rm**: "remove". This command is used to remove files and folders from the terminal. NOTE: Use this command with caution, since deleted files cannot be recovered easily. 
+    - Example 1: `rm thisisatest.txt` -> this removes the text file you created with `touch` in the last example
+    - Example 2: `rm -r this_is_a_folder` -> the `-r` flag means "recursive", and is necessary to delete folders. This command deletes the folder you created in a previous example. 
+
+# Using Makefiles
+_______________________________________________________
+in makefile:
+	tag:
+		command
+	
+	tag1:
+		command $(ARG)
+
+terminal: (w/o arguments)
+	$ make tag: executes commands under tag
+
+terminal: (w/ arguments)
+	$ make tag1 ARG=<VALUE>: executes commands under tag1 w/ specified arguments
+
+# Installing Packages
+_______________________________________________________
+apt install <PACKAGE>: installs <PACKAGE>
+apt update: updates software (run before you install anything)
+apt upgrade: installs newer versions of existing packages
+
+brew install <PACKAGE>: installs <PACKAGE> for macOS only
+
+# Symbols/Keywords
+_______________________________________________________
+- "/": root directory
+- "~": user home directory
+- ".": current working directory
+- "..": superdirectory of current working directory
+- "*": all files w/i working directory
+- "*.ext": all files w/i working directory w/ extension .ext
+- ">": writes the command to an output file
+- "<": reads input file into command
+- "sudo": run command as root user
+
+# Integrate this reference sheet w/ terminal
+_______________________________________________________
+1. Open linux terminal
+2. $cd ~
+3. $touch commands.txt
+4. vim commands.txt
+5. paste this text into commands.txt using vim commands
```

Comment:
to talk about vim, you'd need to explain that first + give the reader instructions on how to install and tell them what exactly vim is